src/utils/FileDownload.ts

```diff
@@ -0,0 +1,16 @@
+export const fileDownload = (data: string | ArrayBuffer | ArrayBufferView | Blob, filename: string, mime?: string) => {
+    const blob = new Blob([data], { type: mime || 'application/octet-stream' });
```

Comment:
I appreciate the test coverage you've added for this!  Thank you!

Do we care about the mime type?  If the point of this code is to trigger "download" capability, 'application/octet-stream' will work in all cases because the browser will not be able to render the Blob and will always prompt to "[Save As](https://developer.mozilla.org/en-US/docs/Web/HTTP/MIME_types#applicationoctet-stream)"

Looks like DAA.js and DAR.js don't pass mime type, so in all cases of application usage, It is effectively "application/octet-stream'

Reply:
I rewrote it like this to keep maximum compatibility with the `js-file-download` package, without including the deprecated bits (like `bom`)